### PR TITLE
[catloop] Connection Establishment Does Not Cancel Inflight Pop Correctly

### PR DESCRIPTION
**Pull request by ppenna**
## Description

This PR is a workaround for https://github.com/demikernel/demikernel/issues/769

### CODE REVIEW

**Review by iyzhang — APPROVED** [review]
LGTM

File: src/rust/catloop/futures/connect.rs [hunk 1]
```diff
@@ -279,9 +302,19 @@ fn connect_ack_received(
             duplex_pipe,
         };
     } else {
-        // Connection timeout, retry.
-        DuplexPipe::drop(&self_.catmem, qt_rx)?;
-        self_.state = ClientState::InitiateConnectRequest;
+        if attempt > MAX_CONNECT_ATTEMPTS {
```
Comment (anandbonde) [inline src/rust/catloop/futures/connect.rs:305]: This name is a bit confusing. It's unclear at a glance whether it corresponds to `InitiateConnectRequest` or `ConnectAckReceived`. Is there a way to simplify this? May be naming the constant appropriately? (e.g. MAX_CONNECT_ACK_RECEIPT_ATTEMPTS / MAX_ACK_RECEIVED_ATTEMPTS etc.)

**Review by anandbonde — APPROVED** [review]
LGTM

File: src/rust/catloop/futures/connect.rs (hunk 1 above)
Comment (ppenna) [reply src/rust/catloop/futures/connect.rs]: Sure.

**Pull request merged**